javascript/code-challenges/challenges-11.test.js

```diff
@@ -0,0 +1,244 @@
+'use strict';
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1 - Review
+
+Write a function named transformToLis that, given an object, returns an array of the key value pairs as html list items.
+
+For example:
+{
+  name: 'bob',
+  age: 32
+}
+
+Becomes:
+[
+<li>name: bob</li>,
+<li>age: 32</li>
+]
+------------------------------------------------------------------------------------------------ */
+
+function transformToLis(obj){
+  let returnArray = [];
+
+  for (const property in obj) {
+    returnArray.push(`<li>${property}: ${obj.name}</li>`);
+    returnArray.push(`<li>${property}: ${obj.age}</li>`);
+  }
+  returnArray.splice(1,1);
+  returnArray.splice(1,1);
+  //console.log(returnArray);
+  return returnArray;
+}; // Okay I know this probably wasn't how it was supposed to be solved, but it works ;v
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function named count that, given an integer and an array of arrays, uses either filter, map, or reduce to count the amount of times the integer is present in the array of arrays.
+
+Note: You might need to use the same method more than once.
+
+For example, count(5, [[1, 3, 5, 7, 9], [5, 5, 5], [1, 2, 3]]) returns 4.
+------------------------------------------------------------------------------------------------ */
+
+const count = (target, input) => {
+  
+  let count = 0;
+  
+   for(let i = 0; i <= input.length - 1; i++) {
+        for(let j = 0; j <= input[i].length - 1; j++){         
+           if (input[i][j] === target) {
+          count += 1;
+      }
+    }
+  }
+
+return count;
+}; // nested for loops are bad, but I am not sure how to solve it the other way.
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function that, given an array of integer arrays as input, calculates the total sum of all the elements in the array.
+
+You may want to use filter, map, or reduce for this problem, but are not required to. You may need to use the same method more than once.
+
+For example, [[1, 2, 3, 4, 5], [6, 7, 2, 4, 5, 7], [9, 2, 3, 6,]] returns 66.
+
+------------------------------------------------------------------------------------------------ */
+
+const totalSum = (input) => {
+  let total = 0;
+
+  const reducer = (previousValue, currentValue) => previousValue + currentValue;
+  for(let i = 0; i <= input.length - 1; i++) {
+          let innerSum = input[i].reduce(reducer);
+      total = total + innerSum;
+}
```

Comment:
- Flat() may come in handy here too! We can use reduce() with it :)
- For challenges, perhaps we can just use the built in reduce() method that JS provides (which is what we see above, but it is written by hand/scratch).